cdap-ui/app/cdap/components/Lab/index.tsx

```diff
@@ -58,34 +60,48 @@ const styles = (): StyleRules => {
 interface ILabProps extends WithStyles<typeof styles> {}
 interface IExperiment {
   id: string;
-  value: boolean;
+  enabled: boolean;
   screenshot: string | null;
   name: string;
   description: string;
+  showValue?: boolean;
+  valueLabel?: string;
+  valueType?: string;
 }
 interface ILabState {
   experiments: IExperiment[];
 }
 
 class Lab extends React.Component<ILabProps, ILabState> {
-  public componentWillMount() {
-    experimentsList.forEach((experiment) => {
-      experiment.value = window.localStorage.getItem(experiment.id) === 'true' ? true : false;
+  public componentDidMount() {
+    experimentsList.forEach((experiment: IExperiment) => {
+      // If experiment preference is present in storage, use it.
+      // If not, use the default value and set it in storage and use it.
+      const experimentStatusFromStorage = window.localStorage.getItem(experiment.id);
+      if (experimentStatusFromStorage === null) {
+        window.localStorage.setItem(experiment.id, experiment.enabled.toString());
+      } else {
+        experiment.enabled = experimentStatusFromStorage === 'true' ? true : false;
```

Comment:
nit: We don't need the ternary operator here (the equality should do the same thing).